keanu-project/src/main/java/io/improbable/keanu/network/BayesianNetwork.java

```diff
@@ -18,15 +15,35 @@
 public class BayesianNetwork {
 
     private final List<? extends Vertex> vertices;
+    private final Map<String, Vertex> vertexLabels;
```

Comment:
Using`String` for the key will lead to issues around typos, capitalisation, white space, and type safety. By encapsulating that in a class we can avoid all those problems by overriding `equals` and `hashCode`.